### PR TITLE
UINV-557: Improve error message when attempting to save an invoice line when no budget for the fiscal year exists

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UINV-557](https://folio-org.atlassian.net/browse/UINV-557): Improve error message when attempting to save an invoice line when no budget for the fiscal year exists
## Approach

https://github.com/user-attachments/assets/b5f42d7c-40ac-4c55-8c99-3a0647d37324

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by alisher-epam** [comment]
@usavkov-epam  @folio-org/fe-tl-reviewers please review

File: src/invoices/InvoiceForm/InvoiceForm.test.js [hunk 1]
```diff
@@ -126,6 +126,7 @@ describe('InvoiceForm component', () => {
     const accountLabel = `${accountNo} (${appSystemNo})`;
 
     expect(container.querySelector('#selected-accounting-code-selection-item')).toHaveTextContent('');
+    await userEvent.click(screen.getAllByText('stripes-components.selection.controlLabel')[4]);
```
Comment (usavkov-epam) [inline src/invoices/InvoiceForm/InvoiceForm.test.js:129]: Is there another way to bind to an element other than the index in an array of elements of the same type? This is a fairly brittle method that can break tests when new elements are added.

Applicable for all places in this file

File: src/invoices/InvoicesList/InvoicesListFilters/FiscalYearFilter/FiscalYearFilter.test.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ describe('FiscalYearFilter', () => {
     renderFilter();
 
     expect(screen.getByText(labelId)).toBeInTheDocument();
+    fireEvent.click(screen.getByText('stripes-components.selection.controlLabel'));
```
Comment (usavkov-epam) [inline src/invoices/InvoicesList/InvoicesListFilters/FiscalYearFilter/FiscalYearFilter.test.js:36]: please use the "userEvent" instead from the corresponding library

File: src/invoices/InvoiceDetails/utils.js [hunk 3]
```diff
@@ -135,19 +139,27 @@ export const showUpdateInvoiceError = async (
...
       }
 
       if (fundId) {
         fundMutator.GET({ path: `${FUNDS_API}/${fundId}` })
-          .then(({ fund }) => {
+          .then(async ({ fund }) => {
+            let fiscalYear = {};
+
+            if (fiscalYearId) {
+              fiscalYear = await ky.get(`${FISCAL_YEARS_API}/${fiscalYearId}`).json();
```
Comment (usavkov-epam) [inline src/invoices/InvoiceDetails/utils.js:154]: The request can be unsuccessful.

**Review by usavkov-epam — COMMENTED** [review]
The PR has quality gates issue: https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=809&id=org.folio%3Aui-invoice&open=AZIzCX26_UrPZE2OH8d2

This code is becoming increasingly bulky and challenging to maintain. In my view, the logic should be moved out to separate "resolver" functions that will handle errors for a specific error code, and then the switch-case construction will map the code to a specific resolver.

E.g.
```
case ERROR_CODES.budgetNotFoundByFundIdAndFiscalYearId: {
  await handleSomeError(<data>);
  break;
}
```

File: src/invoices/InvoiceDetails/utils/utils.js [hunk 4]
```diff
@@ -179,6 +139,7 @@ export const handleInvoiceLineErrors = async ({
   requestData = [],
   responses = [],
   showCallout,
+  ky = noop,
```
Comment (usavkov-epam) [inline src/invoices/InvoiceDetails/utils/utils.js:142]: `noop.get()` -> `TypeError: noop.get is not a function`. `ky` it's an object, not the function

**Review by usavkov-epam — APPROVED** [review]
Except the comment

File: src/invoices/InvoiceForm/InvoiceForm.test.js (hunk 1 above)
Comment (usavkov-epam) [reply src/invoices/InvoiceForm/InvoiceForm.test.js]: And this not resolved as well. Please use specific anchors for selections like `screen.getByRole('button', { name: <name> })`

File: src/invoices/InvoiceDetails/utils/utils.js (hunk 4 above)
Comment (alisher-epam) [reply src/invoices/InvoiceDetails/utils/utils.js]: updated

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**